### PR TITLE
Automation: Add support for job result data

### PR DESCRIPTION
**Pull request by psiinon**
The job data isnt actually used in this add-on its used in reporting, but we'll need to release this add-on before I can raise the reporting PR :)

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Review by kingthorin — APPROVED** [review]

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJob.java [hunk 1]
```diff
@@ -165,6 +166,22 @@ public void runJob(
...
+    @Override
+    public List<JobResultData> getJobResultData() {
+        ActiveScan lastScan = this.getExtAScan().getLastScan();
+        if (lastScan != null) {
+            return getJobResultData(lastScan.getId());
+        }
+        return new ArrayList<JobResultData>();
+    }
+
+    private List<JobResultData> getJobResultData(int scanId) {
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJob.java:181]: `createJobResultData`

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJobResultData.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    public Collection<RuleData> getAllRuleData() {
+        return ruleDataMap.values();
+    }
+
+    @Override
+    public String getKey() {
+        return "activeScanData";
+    }
+
+    public class RuleData {
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJobResultData.java:65]: `static`

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJobResultData.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        private int id;
+        private String name;
+        private long timeTakenMs = 0;
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJobResultData.java:68]: The initialisation could be removed.

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJobResultData.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    @Override
+    public String getKey() {
+        return "activeScanData";
+    }
+
+    public class RuleData {
+        private int id;
+        private String name;
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJobResultData.java:67]: Could be final.

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/PassiveScanJobResultData.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    public Collection<RuleData> getAllRuleData() {
+        return ruleDataMap.values();
+    }
+
+    @Override
+    public String getKey() {
+        return "passiveScanData";
+    }
+
+    public class RuleData {
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/PassiveScanJobResultData.java:71]: Same comments as for the active one.

**Comment by psiinon** [comment]
Thanks - comments hopefully addressed.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**